terraform/modules/gke/gke.tf

```diff
@@ -0,0 +1,71 @@
+variable "infra_name" {
+  type = string
+}
+
+variable "resource_prefix" {
+  type = string
+}
+
+variable "gcp_region" {
+  type = string
+}
+
+variable "machine_type" {
+  type = string
+}
+
+data "google_compute_zones" "available" {}
+
+resource "google_container_cluster" "cluster" {
+  name = "${var.resource_prefix}-cluster"
+  # Specifying a region and not a zone here gives us a regional cluster, meaning
+  # we get cluster masters across multiple zones. This could be overkill for our
+  # availability requirements.
+  location = var.gcp_region
+  node_locations = [
+    data.google_compute_zones.available.names[0],
+    data.google_compute_zones.available.names[1],
+    data.google_compute_zones.available.names[2]
+  ]
+  description = "Prio data share processor ${var.infra_name}"
+  # https://www.terraform.io/docs/providers/google/r/container_cluster.html#remove_default_node_pool
+  remove_default_node_pool = true
+  initial_node_count       = 1
+}
+
+resource "google_container_node_pool" "worker_nodes" {
+  name       = "${var.resource_prefix}-node-pool"
+  location   = var.gcp_region
+  cluster    = google_container_cluster.cluster.name
+  node_count = 1
+  autoscaling {
+    min_node_count = 1
+    max_node_count = 3
+  }
+  management {
+    auto_repair  = true
+    auto_upgrade = true
+  }
+  node_config {
```

Comment:
I assume that by control plane here you mean our execution manager rather than the Kubernetes scheduler GKE runs for us. I agree with your analysis here. It should be easy to add another `google_container_node_pool` for the control plane nodes that's not preemptible, but I think we can discuss that as part of the execution manager implementation.

Reply:
I was thinking of the k8s "system jobs" that need to run in order to mange in-cluster operations, e.g.:
![image](https://user-images.githubusercontent.com/341401/94747985-14ad8c00-0335-11eb-909e-c718b94d5286.png)